pandas/core/indexes/timedeltas.py

```diff
@@ -180,12 +194,23 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
         return subarr
 
     @classmethod
-    def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        result = super(TimedeltaIndex, cls)._simple_new(values, freq, **kwargs)
+    def _simple_new(cls, values, name=None, freq=None, dtype=_TD_DTYPE):
+        # `dtype` is passed by _shallow_copy in corner cases, should always
+        #  be timedelta64[ns] if present
+        assert dtype == _TD_DTYPE
+
+        assert isinstance(values, np.ndarray), type(values)
+        if values.dtype == 'i8':
+            values = values.view('m8[ns]')
+        assert values.dtype == 'm8[ns]', values.dtype
```

Comment:
Same questions for all your assert statements:

* Are these internal? If not, it would be nice to have user-friendly error messages.
* Can these assert statements be tested in any way?

Reply:
The question of testing is actually more general to all of these changes.  Even though it's been labeled as internal, not sure if any of it surfaces in any way.

Reply:
> The edits in simplenew are thoroughly internal; users shouldn’t get near it.
>
> Not sure what testing these assertions would look like. Can you elaborate what you have in m > mind?

(moving your responses to the conversation bubble in the UI, organizational thing)

@jbrockmendel : What I was wondering was whether we could trigger these `assert` statements (e.g. an invalid input to a publicly facing function or method)?

Might be tricky if these edits are purely internal, and if it is too difficult, not a big deal.  Just out of curiosity since tests are good if we can have them.

Reply:
I see.  Yah, we could
Write tests where we directly pass invalid inputs to simplenew. (Typing with thumbs, feel free to reformat if necessary)